### DISCUSSION TITLE
EMS-ESP, DS18B20

### DISCUSSION THREAD

**Discussion by bluhminga** [emsesp/EMS-ESP32#1325] [Q&A]
Hello,

I use EMS-ESP with the "BBQ interface card V3.1". And an "az delivery esp32 wroom 32" module.
Works great in conjunction with Home Assistant.

Is it possible to connect a DS18B20 to the ESP32 module and display it in EMS-ESP? And then send the “room temperature” to Home Assistant?

Thanks in advance.

**Comment by proddy** [comment]
yes, just add to an available GPIO, set the GPIO in EMS-ESP Setting's and it'll be automatically discovered. Don't forget to add a resistor.

**Reply by bluhminga** [reply]
Hello,

"add an available GPIO, set the GPIO in the EMS-ESP settings" that's exactly what I tried. Without success.

Is there a specific GPIO? Or, where in the EMS-ESP will I find the sensor?

**Reply by proddy** [reply]
Look at the manufacture's recommendations for your ESP32 (az delivery?). Generally any input GPIO will work. There are listed on https://emsesp.github.io/docs/Configuring/#network-settings-network-network-settings

The Sensor will show in the Sensor tab under the Dashboard. You can see an example at https://ems-esp.derbyshire.nl/dashboard/sensors
